### DISCUSSION TITLE
Question about structuring configuration

### DISCUSSION THREAD

**Discussion by dpetka2001** [LazyVim/LazyVim#140] [Q&A] [answered]
So let's say for example that I want to pass some options to Treesitter and I also want to extend the `ensure_installed`
parser configuration. How would I go about doing that? I know I must pass a table to `opts` to change some options and 
I know I must pass a function to `opts` to `vim.extend_list` the `ensure_installed` list. Should I have 2 different blocks of 
treesitter configuration in my file?
Example:
```lua
  {
    "nvim-treesitter",
    opts = function(_, opts)
      vim.list_extend(opts.ensure_installed, {
        "c",
        "cpp",
        "fish",
        "rust",
      })
    end,
  },
``` 
and 
```lua
  {
    "nvim-treesitter",
    opts = {
      my_options_here
    },
  },
```
Or is there a way to combine both of them in a single spec block?
Forgive my ignorance as I'm just a coding noob.

**Comment by folke** [comment, marked as answer]
You could do it in two blocks like you did, or in one like this:

```lua
return {
	"nvim-treesitter",
	opts = function(_, opts)
		vim.list_extend(opts.ensure_installed, {
			"c",
			"cpp",
			"fish",
			"rust",
		})
    return vim.tbl_deep_extend("force", opts, {
      option1 = true,
      option2 = true
    })
	end,
}
```

**Reply by dpetka2001** [reply]
Thank you very much for your help!!
